src/main/kotlin/org/axonframework/intellij/ide/plugin/support/AxonErrorReportSubmitter.kt

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2022. Axon Framework
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.axonframework.intellij.ide.plugin.support
+
+import com.intellij.diagnostic.AbstractMessage
+import com.intellij.ide.DataManager
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.diagnostic.ErrorReportSubmitter
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent
+import com.intellij.openapi.diagnostic.SubmittedReportInfo
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.ui.Messages
+import com.intellij.util.Consumer
+import java.awt.Component
+
+class AxonErrorReportSubmitter : ErrorReportSubmitter() {
+    private val service = ApplicationManager.getApplication().getService(ReportingService::class.java)
```

Comment:
Curious, but when will IntelliJ decide to invoke this `ErrorReportSubmitter` implementation?

Reply:
When an exception is thrown of which the originating caller is in our classloader. Every plugin has its own classloader, so IntelliJ knows which plugin causes the exception.

It will take the ErrorReportsubmitter defined by the extensionpoint, and otherwise their own. The button will then be greyed out though